lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,545 @@
+        if result_state:
+            state_condition = lambda entity: entity and entity.status == result_state
+        wait(
+            service=entity_service,
+            condition=state_condition,
+            fail_condition=fail_condition,
+            wait=self._module.params['wait'],
+            timeout=self._module.params['timeout'],
+        )
+
+        return {
+            'changed': self.changed,
+            'id': entity.id,
+            type(entity).__name__.lower(): get_dict_of_struct(entity),
+        }
+
+    def pre_remove(self, entity):
+        """
+        This method is called right before entity is removed.
+
+        :param entity: Entity which we want to remove.
+        """
+        pass
+
+    def remove(self, entity=None, search_params=None, **kwargs):
+        """
+        Method which is called when state of the entity is 'absent'. If user
+        don't provide `entity` parameter the entity is searched using
+        `search_params` parameter. If entity is found it's removed.
+
+        Additional task can be executed after entity is searched and before
+        remove action is executed by overriding `pre_remove` method.
+
+        :param entity: Entity we want to remove.
+        :param search_params: Dictionary of parameters to be used for search.
+        :param kwargs: Additional parameters passed when removing entity.
+        :return: Dictionary with values returned by Ansible module.
+        """
+        if entity is None:
+            entity = self.search_entity(search_params)
+
+        if entity is None:
+            return {
+                'changed': self.changed,
+                'msg': "Entity wasn't found."
+            }
+
+        self.pre_remove(entity)
+
+        entity_service = self._service.service(entity.id)
+        if not self._module.check_mode:
+            entity_service.remove(**kwargs)
+
+            wait(
+                service=entity_service,
+                condition=lambda entity: not entity,
+                wait=self._module.params['wait'],
+                timeout=self._module.params['timeout'],
+            )
+        self.changed = True
+
+        return {
+            'changed': self.changed,
+            'id': entity.id,
+            type(entity).__name__.lower(): get_dict_of_struct(entity),
+        }
+
+    def action(
+        self,
+        action,
+        entity=None,
+        action_condition=lambda e: e,
+        wait_condition=lambda e: e,
+        fail_condition=lambda e: False,
+        pre_action=lambda e: e,
+        post_action=lambda e: None,
+        search_params=None,
+        **kwargs
+    ):
+        """
+        This method is executed when we want to change the state of some oVirt
+        entity. The action to be executed on oVirt service is specified by
+        `action` parameter. Whether the action should be executed can be
+        specified by passing `action_condition` parameter. Which state the entity
```

Comment:
Which state the entity should have after executing the action can be specified by passing `wait_condition` parameter.

-> 

State which the entity should be in after execution of the action can be specified by `wait_condition` parameter.